docs/design/pce/design.md

```diff
@@ -0,0 +1,57 @@
+# Path Computation Engine (PCE)
+
+## Overview
+
+PCE is a open-kilda module responsible for any network path computations. The main goal is to provide simple API to calculate all paths required for any system component.
+
+## API
+
+All components should interact with PCE through PathComputer interface.
+PathComputer provides following options:
+* Calculate best path for flow
+
+Criteria for best path computation is dependent on flow **path computation strategy** param. For now two options are available: cost and latency. Only links with appropriate **flow encapsulation type** and sufficient bandwidth can be used as available network resources. **Ignore bandwidth flag** may be used to turn off bandwidth limitation. Additionally **flow group id** param can be used to create [diverse flows](../solutions/pce-diverse-flows/pce-diverse-flows.md).
+
+* Calculate path for flow with ability to reuse provided network resources
+
+The same as previous but available network resources are extended with the list of paths which resources may be reused. 
+
+* Calculate list of best available paths
+
+Criteria for best path computation is dependent on **path computation strategy** param. **Encapsulation type** param is used to calculate available network resources in the same way. Best paths are sorted in descending order of MinAvailableBandwidth.
+
+## Algorithm
+
+Breadth-first search with path weight calculation and path length limitation is used to calculate best path. 
+
+Starting from source node we visit all neighbor nodes and saving path to all found nodes with lowest total weight. 
+
+Algorithm pseudocode:
+1. add source node to visit queue
+2. for each node in visit queue
+    1. if node is destination
+        1. if path weight is lower then current best 
+            1. save new path as current best path
+        2. skip further processing of the current node
```

Comment:
nit: this means else? I'd make it explicit here and below

Reply:
No, just next step in "then" block. Added explicit keywords.